File: manager/src/main/java/com/gmail/pzalejko/invoice/manager/domain/invoice/domain/company/BankAccountNumber.java
package com.gmail.pzalejko.invoice.manager.domain.invoice.domain.company;

import com.gmail.pzalejko.invoice.manager.domain.common.ValueObject;
import com.google.common.base.Preconditions;
import lombok.NonNull;

public record BankAccountNumber(@NonNull String value) implements ValueObject {

    public static final BankAccountNumber MISSING = new BankAccountNumber("0".repeat(26));

    public BankAccountNumber {
        Preconditions.checkArgument(value.length() == 26, "Bank account has an invalid length");
    }
}

File: web/src/main/kotlin/com/gmail/pzalejko/invoice/core/model/subject/BankAccountNumber.kt
package com.gmail.pzalejko.invoice.core.model.subject

import com.gmail.pzalejko.invoice.common.ddd.ValueObject

data class BankAccountNumber(val number: String) : ValueObject{

    init {
        require(number.length != 26) { "Bank account has an invalid length" }
    }
}